README.md

```diff
@@ -116,6 +116,20 @@ use the `artifact-name` parameter:
     artifact-name: sbom.spdx
 ```
 
+> [!IMPORTANT]  
+> If using this action within a **matrix build**, you must specify a unique `artifact-name`
+> based on matrix parameters or the artifact upload will fail due to duplicate names. See
+> an [example here](.github/workflows/test.yml#L36).
+
+## Permissions
+
+This action needs the following permissions:
+
+```
+contents: write # for sbom-action artifact uploads
```

Comment:
I'm a bit confused by these YAML comments. It might help if there was a list of use case to required permissions. I suspect there are some permissions always needed, and some only needed for certain use cases.

Reply:
I'm open to suggestions how to make this more clear. There are 2 main things this does: create an SBOM and upload it as a workflow artifact (the first permission) and attach release assets (the second permission), a user only needs to set the permission for the thing they're doing, which the comments indicate.

Reply:
How does `actions: read` grant permissions to attach things to the release assets? Does it not need some write permissions for that?

Reply:
Updated -- is this more clear, @willmurphyscode ?